app/models/cart_item.rb

```diff
@@ -1,2 +1,4 @@
 class CartItem < ApplicationRecord
+  has_many :users, dependent: :destroy
+  has_many :product, dependent: :destroy
```

Comment:
```suggestion
  has_many :product, dependent: :destroy
```
sが抜けてる→has_many :products